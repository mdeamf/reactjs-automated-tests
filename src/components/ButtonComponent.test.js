// Link.react.test.js
import React from 'react';
import renderer from 'react-test-renderer';
import Button from './ButtonComponent';

test('Button changes when clicked', () => {
  const {act} = renderer;
  const component = renderer.create(<Button title="My snapshot button!" />);
  let tree = component.toJSON();
  expect(tree).toMatchSnapshot();

  act(() => {
    tree.props.onClick();
  });
  tree = component.toJSON();
  expect(tree).toMatchSnapshot();

  act(() => {
    tree.props.onClick();
  });
  tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});
