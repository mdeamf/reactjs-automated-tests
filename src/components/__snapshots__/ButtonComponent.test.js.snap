// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button changes when clicked 1`] = `
<button
  onClick={[Function]}
  style={
    Object {
      "color": "blue",
    }
  }
>
  My snapshot button!
</button>
`;

exports[`Button changes when clicked 2`] = `
<button
  onClick={[Function]}
  style={
    Object {
      "color": "red",
    }
  }
>
  My snapshot button!
</button>
`;

exports[`Button changes when clicked 3`] = `
<button
  onClick={[Function]}
  style={
    Object {
      "color": "blue",
    }
  }
>
  My snapshot button!
</button>
`;
